algorithms/unit_tests/TestStdAlgorithmsTeamSearchN.cpp

```diff
@@ -0,0 +1,301 @@
+  // value that is not present in dataView
+
+  const std::size_t halfCols = (numCols > 1) ? ((numCols + 1) / 2) : (1);
+  const std::size_t seqSize  = (numCols > 1) ? (std::log2(numCols)) : (1);
+
+  Kokkos::View<ValueType*> searchedValuesView("searchedValuesView", numTeams);
+  auto searchedValuesView_h = create_host_space_copy(searchedValuesView);
+
+  // dataView might not deep copyable (e.g. strided layout) so to prepare it
+  // correclty, we make a new view that is for sure deep copyable, modify it
+  // on the host, deep copy to device and then launch a kernel to copy to
+  // dataView
+  auto dataView_dc =
+      create_deep_copyable_compatible_view_with_same_extent(dataView);
+  auto dataView_dc_h = create_mirror_view(Kokkos::HostSpace(), dataView_dc);
+
+  if (sequencesExist) {
+    const std::size_t dataBegin = halfCols - seqSize;
+    for (std::size_t i = 0; i < searchedValuesView.extent(0); ++i) {
+      const ValueType searchedVal = dataView_dc_h(i, dataBegin);
+      searchedValuesView_h(i)     = searchedVal;
+
+      for (std::size_t j = dataBegin + 1; j < seqSize; ++j) {
+        dataView_dc_h(i, j) = searchedVal;
+      }
+    }
+
+    // copy to dataView_dc and then to dataView
+    Kokkos::deep_copy(dataView_dc, dataView_dc_h);
+
+    CopyFunctorRank2 cpFun(dataView_dc, dataView);
+    Kokkos::parallel_for("copy", dataView.extent(0) * dataView.extent(1),
+                         cpFun);
+  } else {
+    using rand_pool =
+        Kokkos::Random_XorShift64_Pool<Kokkos::DefaultHostExecutionSpace>;
+    rand_pool pool(lowerBound * upperBound);
+    Kokkos::fill_random(searchedValuesView_h, pool, upperBound, upperBound * 2);
+  }
+
+  Kokkos::deep_copy(searchedValuesView, searchedValuesView_h);
+
+  // -----------------------------------------------
+  // launch kokkos kernel
+  // -----------------------------------------------
+  using space_t = Kokkos::DefaultExecutionSpace;
+  Kokkos::TeamPolicy<space_t> policy(numTeams, Kokkos::AUTO());
+
+  // search returns an iterator so to verify that it is correct each team stores
+  // the distance of the returned iterator from the beginning of the interval
+  // that team operates on and then we check that these distances match the std
+  // result
+  Kokkos::View<std::size_t*> distancesView("distancesView", numTeams);
+
+  EqualFunctor<ValueType> binaryOp;
+
+  // use CTAD for functor
+  TestFunctorA fnc(dataView, seqSize, searchedValuesView, distancesView,
+                   binaryOp, apiId);
+  Kokkos::parallel_for(policy, fnc);
+
+  // -----------------------------------------------
+  // run cpp-std kernel and check
+  // -----------------------------------------------
+  auto distancesView_h = create_host_space_copy(distancesView);
+
+  for (std::size_t i = 0; i < dataView.extent(0); ++i) {
+    auto rowFrom = Kokkos::subview(dataView_dc_h, i, Kokkos::ALL());
+
+    const auto rowFromBegin = KE::begin(rowFrom);
+    const auto rowFromEnd   = KE::end(rowFrom);
```

Comment:
just a small detail, but i would make these cbeging and cend since we only need read access

Reply:
Done.